Assets/HoloToolkit/UX/Scripts/Tooltip/ToolTipConnector.cs

```diff
@@ -0,0 +1,350 @@
+    {
+        [SerializeField]
+        private ToolTip toolTip;
+
+        [SerializeField]
+        private ConnectorFollowType connectorFollowType = ConnectorFollowType.AnchorOnly;
+        /// <summary>
+        /// getter /setter for the follow style of the tooltip connector
+        /// </summary>
+        public ConnectorFollowType FollowingType
+        {
+            get
+            {
+                return connectorFollowType;
+            }
+
+            set
+            {
+                connectorFollowType = value;
+            }
+        }
+
+        [SerializeField]
+        private ConnnectorPivotMode pivotMode = ConnnectorPivotMode.Manual;
+        /// <summary>
+        /// is the connector pivot set manually or automatically
+        /// </summary>
+        public ConnnectorPivotMode PivotingMode
+        {
+            get
+            {
+                return pivotMode;
+            }
+
+            set
+            {
+                pivotMode = value;
+            }
+        }
+
+        [SerializeField]
+        private ConnectorPivotDirection pivotDirection = ConnectorPivotDirection.North;
+        /// <summary>
+        /// getter/setter for the direction of the connector
+        /// </summary>
+        public ConnectorPivotDirection PivotDirection
+        {
+            get
+            {
+                return pivotDirection;
+            }
+
+            set
+            {
+                pivotDirection = value;
+            }
+        }
+
+        [SerializeField]
+        private ConnectorOrientType pivotDirectionOrient = ConnectorOrientType.OrientToObject;
+        /// <summary>
+        /// orientation style for connector
+        /// </summary>
+        public ConnectorOrientType PivotDirectionOrient
+        {
+            get
+            {
+                return pivotDirectionOrient;
+            }
+
+            set
+            {
+                pivotDirectionOrient = value;
+            }
+        }
+
+        [SerializeField]
+        private Vector3 manualPivotDirection = Vector3.up;
+        /// <summary>
+        /// getter/setter for direction of pivot manually set
+        /// </summary>
+        public Vector3 ManualPivotDirection
+        {
+            get
+            {
+                return manualPivotDirection;
+            }
+
+            set
+            {
+                manualPivotDirection = value;
+            }
+        }
+
+        [SerializeField]
+        private Vector3 manualPivotLocalPosition = Vector3.up;
+        /// <summary>
+        /// getter/setter for local pivot position
+        /// </summary>
+        public Vector3 ManualPivotLocalPosition
+        {
+            get
+            {
+                return manualPivotLocalPosition;
+            }
+
+            set
+            {
+                manualPivotLocalPosition = value;
+            }
+        }
+
+        [SerializeField]
+        [Range(0f, 2f)]
+        private float pivotDistance = 0.25f;
```

Comment:
does this ever get set internally by code? if so, consider adding a range check setter